From 2bf963605442b2693c7adc06f37957984e0ced56 Mon Sep 17 00:00:00 2001
From: Andreas Enge <andreas.enge@inria.fr>
Date: Mon, 12 Dec 2022 11:46:31 +0100
Subject: [PATCH] Simplify tdummy test.

* tests/tdummy.c: Include mpc.h without stdio.h; the test always succeeds
  at run time, but may fail during compilation if there is an error in
  mpc.h.
* tests/mpc-tests.h: Drop DONT_INCLUDE_STDIO macro.
---
 tests/mpc-tests.h | 6 ------
 tests/tdummy.c    | 7 +------
 2 files changed, 1 insertion(+), 12 deletions(-)

diff --git a/tests/mpc-tests.h b/tests/mpc-tests.h
index c6b5679..ecd10b7 100644
--- a/tests/mpc-tests.h
+++ b/tests/mpc-tests.h
@@ -22,9 +22,7 @@ along with this program. If not, see http://www.gnu.org/licenses/ .
 #define __MPC_TESTS_H
 
 #include "config.h"
-#ifndef DONT_INCLUDE_STDIO
 #include <stdio.h>
-#endif
 #include <ctype.h>
 #include <stdlib.h>
 #include <inttypes.h> /* for the PRIi64 format modifier */
@@ -141,7 +139,6 @@ extern int same_mpc_value  (mpc_ptr got, mpc_ptr ref,
                             known_signs_t known_signs);
 
 /** READ FILE WITH TEST DATA SET **/
-#ifndef DONT_INCLUDE_STDIO
 extern FILE * open_data_file         (const char *file_name);
 extern void   close_data_file        (FILE *fp);
 
@@ -156,7 +153,6 @@ extern size_t read_string            (FILE *fp, char **buffer_ptr,
                                       size_t buffer_length, const char *name);
 extern void read_mpfr                (FILE *fp, mpfr_ptr x, int *known_sign);
 extern void read_mpc                 (FILE *fp, mpc_ptr z, known_signs_t *ks);
-#endif
 
 void set_mpfr_flags   (int counter);
 void check_mpfr_flags (int counter);
@@ -289,7 +285,6 @@ typedef struct {
 void        read_description    (mpc_fun_param_t* param, const char *file);
 const char* read_description_findname (mpc_param_t e);
 
-#ifndef DONT_INCLUDE_STDIO
 /* file functions */
 typedef struct {
   char *pathname;
@@ -342,7 +337,6 @@ void    tpl_read_mpc        (mpc_datafile_context_t* datafile_context,
                              mpc_data_t* z);
 void    tpl_read_mpc_rnd    (mpc_datafile_context_t* datafile_context,
                              mpc_rnd_t* rnd);
-#endif
 
 int     tpl_same_mpz_value  (mpz_ptr n1, mpz_ptr n2);
 int     tpl_same_mpfr_value (mpfr_ptr x1, mpfr_ptr x2, int known_sign);
diff --git a/tests/tdummy.c b/tests/tdummy.c
index 23072f5..e525609 100644
--- a/tests/tdummy.c
+++ b/tests/tdummy.c
@@ -18,16 +18,11 @@ You should have received a copy of the GNU Lesser General Public License
 along with this program. If not, see http://www.gnu.org/licenses/ .
 */
 
-#define DONT_INCLUDE_STDIO
-#include "mpc-tests.h"
+#include "mpc.h"
 
 int
 main (void)
 {
-  test_start ();
-
-  test_end ();
-
   return 0;
 }
 
-- 
GitLab

